File: ai-backend/riskAI/domain/Moves/AttackResponse.py
import json


class AttackResponse:
    def __init__(self, amountOfSurvivingTroopsAttacker, amountOfSurvivingTroopsDefender, gameId, attackerDices, defenderDices):
        self.amountOfSurvivingTroopsAttacker = amountOfSurvivingTroopsAttacker
        self.amountOfSurvivingTroopsDefender = amountOfSurvivingTroopsDefender
        self.gameId = gameId
        self.attackerDices = attackerDices
        self.defenderDices = defenderDices

    def __str__(self):
        return "AttackResponse: " + str(self.amountOfSurvivingTroopsAttacker) + " " + str(self.amountOfSurvivingTroopsDefender)

    @classmethod
    def from_json(cls, json_string):
        json_dict = json.loads(json_string)
        return cls(**json_dict)

File: backend/Risk-Backend/src/main/java/kdg/be/riskbackend/game/dtos/phases/AttackResponse.java
package kdg.be.riskbackend.game.dtos.phases;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class AttackResponse {
    private int amountOfSurvivingTroopsAttacker;
    private int amountOfSurvivingTroopsDefender;
    private long gameId;
    private List<Integer> attackerDices;
    private List<Integer> defenderDices;
    private boolean attackerWonGame;

    /**
     * class constructor
     *
     * @param gameId the id of the game
     * @param attackerDices the dices (eyes) of the attacker
     * @param defenderDices the dices (eyes) of the defender
     * @param attackerWonGame true if the attacker won the hole game
     */
    public AttackResponse(long gameId, List<Integer> attackerDices, List<Integer> defenderDices, boolean attackerWonGame) {
        this.attackerWonGame = attackerWonGame;
        this.gameId = gameId;
        this.attackerDices = attackerDices;
        this.defenderDices = defenderDices;
    }
}
